salt/modules/state.py

```diff
@@ -1880,6 +1880,57 @@ def show_sls(mods, test=None, queue=False, **kwargs):
     return high_
 
 
+def sls_exists(mods, test=None, queue=False, **kwargs):
+    '''
+    Tests for the existance the of a specific SLS or list of SLS files on the
+    master. Similar to :py:func:`state.show_sls <salt.modules.state.show_sls>`,
+    rather than returning state details, returns True or False. The default
+    environment is ``base``, use ``saltenv`` to specify a different environment.
+
+    .. versionadded:: Fluorine
+
+    saltenv
+        Specify a salt fileserver environment from which to look for the SLS files
+        specified in the ``mods`` argument
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' state.sls_exists core,edit.vim saltenv=dev
+    '''
+    return isinstance(
+        show_sls(mods, test=test, queue=queue, **kwargs),
+        dict
+    )
+
+
+def id_exists(ids, mods, test=None, queue=False, **kwargs):
+    '''
+    Tests for the existence of a specific ID or list of IDs within the
+    specified SLS file(s). Similar to :py:func:`state.sls_exists
+    <salt.modules.state.sls_exists>`, returns True or False. The default
+    environment is base``, use ``saltenv`` to specify a different environment.
+
+    .. versionadded:: Fluorine
+
+    saltenv
+        Specify a salt fileserver environment from which to look for the SLS files
+        specified in the ``mods`` argument
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' state.id_exists create_myfile,update_template filestate saltenv=dev
+    '''
+    if isinstance(ids, six.string_types):
+        ids = ids.split(',')
```

Comment:
Please use `salt.utils.args.split_input` instead of this:
```python
def id_exists(...
    sls_ids = set(x['__id__'] for x ...
    return set(salt.utils.args.split_input(ids)).issubset(sls_ids)
```
And it would be nice to others not to use one-char variables.

Reply:
I don't mind the 1-char in a comprehension expression, to be honest. I don't think that needs to change.

@isbm is right about the split_input, we do have a helper for this. It could be done as

```python
ids = salt.utils.args.split_input(ids)
```